tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2724,14 +2724,63 @@ def test_on_demand_map_with_multiple_selected(self):
             query=query_str,
             selected_columns=["eps()", "epm()", "not_on_demand"],
             config=QueryBuilderConfig(
-                on_demand_metrics_enabled=True, on_demand_metrics_type=MetricSpecType.SIMPLE_QUERY
+                on_demand_metrics_enabled=True, on_demand_metrics_type=MetricSpecType.DYNAMIC_QUERY
             ),
         )
         assert query._on_demand_metric_spec_map
         assert query._on_demand_metric_spec_map["eps()"]
         assert query._on_demand_metric_spec_map["epm()"]
         assert "not_on_demand" not in query._on_demand_metric_spec_map
 
+    def test_on_demand_map_with_multiple_percentiles(self):
+
+        field = "p75(measurements.fcp)"
+        field_two = "p75(measurements.lcp)"
+        groupbys = []
+        query = "transaction.duration:>=100"
+        spec = OnDemandMetricSpec(
+            field=field, groupbys=groupbys, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY
+        )
+        spec_two = OnDemandMetricSpec(
+            field=field_two, groupbys=groupbys, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY
+        )
+
+        for day in range(0, 4):
+            self.store_on_demand_metric(
+                day * 250,
+                spec=spec,
+                timestamp=self.start + datetime.timedelta(days=0),
```

Comment:
Set day to = 0 here, since for distribution we have to make more than one metric for it to work properly for a time bucket